File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/winforms/samples/data/webservicebinding/cs/client/WebServiceClient.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------


namespace Microsoft.Samples.Windows.Forms.Cs.WebServiceClient {
    using System;
    using System.ComponentModel;
    using System.Drawing;
    using System.Windows.Forms;

    using System.Data;
    using System.Data.OleDb;

    using Microsoft.Samples.Windows.Forms.Cs.WebServiceBinding.localhost;

    using System.Net;
    using System.IO;


    public class WebServiceClient : System.Windows.Forms.Form {
        private System.ComponentModel.Container components;
        private System.Windows.Forms.StatusBar statusBar1;
        private Microsoft.Samples.Windows.Forms.Cs.WebServiceBinding.Data.CustomersAndOrdersDataSet customersAndOrdersDataSet1;
        private System.Windows.Forms.Button buttonLoad;
        private System.Windows.Forms.DataGrid dataGrid1;

        public WebServiceClient() {

            // Required by the Windows Forms Designer
            InitializeComponent();
            if ( dataGrid1.TableStyles.Count == 0 ) {
                dataGrid1.TableStyles.Add(new DataGridTableStyle());
            }
            dataGrid1.TableStyles[0].PreferredRowHeight = 16;
            dataGrid1.TableStyles[0].AlternatingBackColor = System.Drawing.Color.WhiteSmoke;
        }

        protected void buttonLoad_Click(object sender, System.EventArgs e) {
            Cursor currentCursor = Cursor.Current;
            try {
                Cursor.Current = Cursors.WaitCursor;

                statusBar1.Text ="Loading Customers...";

                //Execute the XML Web service to return a DataSet
                CustomersList custList1 = new CustomersList();
                custList1.Credentials =
                   System.Net.CredentialCache.DefaultCredentials;
                DataSet ds1 = custList1.GetCustomers();

                //Merge the new dataset into our customersDataSet
                customersAndOrdersDataSet1.Merge(ds1);

                statusBar1.Text ="Updating Grid...";
            } finally {
                Cursor.Current = currentCursor;
                statusBar1.Text ="Done";
            }
        }


        protected override void Dispose(bool disposing)
        {
           if (disposing) {
                if (components != null) {
                    components.Dispose();
                }
           }
           base.Dispose(disposing);
        }

        private void InitializeComponent() {
            this.components = new System.ComponentModel.Container();
            this.dataGrid1 = new System.Windows.Forms.DataGrid();
            this.statusBar1 = new System.Windows.Forms.StatusBar();
            this.customersAndOrdersDataSet1 = new Microsoft.Samples.Windows.Forms.Cs.WebServiceBinding.Data.CustomersAndOrdersDataSet();
            this.buttonLoad = new System.Windows.Forms.Button();

            dataGrid1.BeginInit();

            dataGrid1.Text = "dataGrid1";
            dataGrid1.Size = new System.Drawing.Size(584, 336);
            dataGrid1.DataSource = customersAndOrdersDataSet1;
            dataGrid1.DataMember = "Customers";
            dataGrid1.ForeColor = System.Drawing.Color.Navy;
            dataGrid1.TabIndex = 0;
            dataGrid1.Anchor = AnchorStyles.Top | AnchorStyles.Bottom | AnchorStyles.Left | AnchorStyles.Right;
            dataGrid1.Location = new System.Drawing.Point(8, 8);
            dataGrid1.BackColor = System.Drawing.Color.Gainsboro;

            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.Text = "Customer Details";
            this.AcceptButton = buttonLoad;
            this.ClientSize = new System.Drawing.Size(600, 413);

            statusBar1.BackColor = System.Drawing.SystemColors.Control;
            statusBar1.Size = new System.Drawing.Size(600, 16);
            statusBar1.TabIndex = 2;
            statusBar1.Text = "Click on Load";
            statusBar1.Location = new System.Drawing.Point(0, 397);

            customersAndOrdersDataSet1.DataSetName = "CustomersDataSet";

            buttonLoad.FlatStyle = System.Windows.Forms.FlatStyle.Flat;
            buttonLoad.Size = new System.Drawing.Size(112, 32);
            buttonLoad.TabIndex = 1;
            buttonLoad.Anchor = AnchorStyles.Bottom | AnchorStyles.Right;
            buttonLoad.Text = "&Load";
            buttonLoad.Location = new System.Drawing.Point(480, 352);
            buttonLoad.Click += new System.EventHandler(buttonLoad_Click);

            this.Controls.Add(statusBar1);
            this.Controls.Add(buttonLoad);
            this.Controls.Add(dataGrid1);

            dataGrid1.EndInit();
        }


        [STAThread]
        public static void Main(string[] args) {
            Application.Run(new WebServiceClient());
        }

    }
}

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/winforms/samples/data/webservicebinding/vb/client/WebServiceClient.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------


Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms

Imports System.Data

Imports Microsoft.Samples.Windows.Forms.Vb.WebServiceBinding.localhost
Imports Microsoft.Samples.Windows.Forms.Vb.WebServiceBinding.Data

Public Class WebServiceClient
    Inherits System.Windows.Forms.Form

    Public Sub New()
        MyBase.New()

        WebServiceClient = Me

        ' Required by the Windows Forms Designer
        InitializeComponent()

        If dataGrid1.TableStyles.Count = 0 Then
            dataGrid1.TableStyles.Add(New DataGridTableStyle())
        End If
        dataGrid1.TableStyles(0).PreferredRowHeight = 16
        dataGrid1.TableStyles(0).AlternatingBackColor = System.Drawing.Color.WhiteSmoke
    End Sub

    'Handle the Load Button Click
    'Load the Customers, Orders and OrderDetails Tables and display in the Grid
    Protected Sub buttonLoad_Click(ByVal sender As Object, ByVal e As System.EventArgs)
        Dim currentCursor As Cursor = Cursor.Current
        Try
            Cursor.Current = Cursors.WaitCursor

            statusBar1.Text = "Loading Customers..."

            'Execute the XML Web service to return a DataSet
            Dim custList1 As New CustomersList()
            custList1.Credentials = System.Net.CredentialCache.DefaultCredentials
            Dim ds1 As DataSet = custList1.GetCustomers()

            'Merge the new dataset into our customersAndOrdersDataSet
            customersAndOrdersDataSet1.Merge(ds1)

            statusBar1.Text = "Updating Grid..."
        Finally
            statusBar1.Text = "Done"
            Cursor.Current = currentCursor
        End Try
    End Sub

    'Clean up any resources being used
    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
        If disposing Then
            If Not (components Is Nothing) Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    End Sub


#Region " Windows Form Designer generated code "

    Private components As System.ComponentModel.Container
    Private WithEvents statusBar1 As System.Windows.Forms.StatusBar
    Private WithEvents buttonLoad As System.Windows.Forms.Button
    Private WithEvents dataGrid1 As System.Windows.Forms.DataGrid
    Private WithEvents customersAndOrdersDataSet1 As Microsoft.Samples.Windows.Forms.Vb.WebServiceBinding.Data.CustomersAndOrdersDataSet

    Private WithEvents WebServiceClient As System.Windows.Forms.Form

    Private Sub InitializeComponent()
        Me.components = New System.ComponentModel.Container()
        Me.dataGrid1 = New System.Windows.Forms.DataGrid()
        Me.statusBar1 = New System.Windows.Forms.StatusBar()
        Me.buttonLoad = New System.Windows.Forms.Button()
        Me.customersAndOrdersDataSet1 = New Microsoft.Samples.Windows.Forms.Vb.WebServiceBinding.Data.CustomersAndOrdersDataSet()

        dataGrid1.BeginInit()

        dataGrid1.Text = "dataGrid1"
        dataGrid1.Size = New System.Drawing.Size(584, 336)
        dataGrid1.ForeColor = System.Drawing.Color.Navy
        dataGrid1.TabIndex = 0
        dataGrid1.Anchor = System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Bottom Or System.Windows.Forms.AnchorStyles.Left Or System.Windows.Forms.AnchorStyles.Right
        dataGrid1.Location = New System.Drawing.Point(8, 8)
        dataGrid1.BackColor = System.Drawing.Color.Gainsboro
        dataGrid1.DataSource = customersAndOrdersDataSet1
        dataGrid1.DataMember = "Customers"

        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        Me.Text = "Customer Details"
        Me.AcceptButton = buttonLoad
        Me.ClientSize = New System.Drawing.Size(600, 413)

        statusBar1.BackColor = System.Drawing.SystemColors.Control
        statusBar1.Size = New System.Drawing.Size(600, 16)
        statusBar1.TabIndex = 2
        statusBar1.Text = "Click on Load"
        statusBar1.Location = New System.Drawing.Point(0, 397)

        buttonLoad.FlatStyle = System.Windows.Forms.FlatStyle.Flat
        buttonLoad.Size = New System.Drawing.Size(112, 32)
        buttonLoad.TabIndex = 1
        buttonLoad.Anchor = System.Windows.Forms.AnchorStyles.Bottom Or System.Windows.Forms.AnchorStyles.Right
        buttonLoad.Text = "&Load"
        buttonLoad.Location = New System.Drawing.Point(480, 352)
        AddHandler buttonLoad.Click, New System.EventHandler(AddressOf buttonLoad_Click)

        Me.Controls.Add(statusBar1)
        Me.Controls.Add(buttonLoad)
        Me.Controls.Add(dataGrid1)

        dataGrid1.EndInit()
    End Sub

#End Region


    'The main entry point for the application
    Shared Sub Main()
        System.Windows.Forms.Application.Run(New WebServiceClient())
    End Sub

End Class
